imapclient/imapclient.py

```diff
@@ -82,6 +83,18 @@ def __new__(cls, personal, other, shared):
     shared = property(itemgetter(2))
 
 
+class SocketTimeout(namedtuple("SocketTimeout", "connect read")):
+    """Represents timeout configuration for an IMAP connection.
+
+    :ivar connect: maximum time to wait for a connection attempt to remote server
+    :ivar read: maximum time to wait for performing a read/write operation
+
+    As an example, ``SocketTimeout(connect=5, read=10)`` will make the socket
```

Comment:
Nitpick but I would change the example to `connect=15, read=60` to provide values that are a good default for real life with slow servers.